lib/rules/no-extra-parens.js

```diff
@@ -402,6 +404,19 @@ module.exports = {
                 if (isIIFE(node) && !isParenthesised(node.callee)) {
                     return;
                 }
+
+                if (ALLOW_PARENS_AFTER_COMMENT_PATTERN && isParenthesised(node)) {
```

Comment:
```suggestion
                if (ALLOW_PARENS_AFTER_COMMENT_PATTERN) {
```

It seems unnecessary to check if `node` is parenthesized, as it's reported because it's parenthesized?

Reply:
Yes, not required.